File: BINDINGS/cxx/eflxx_examples/src/elementaryxx/full/test_radio.cpp
#include "test.h"

void test_radio (void *data, Evas_Object *obj, void *event_info)
{
  Radio *rd = NULL;
  Radio *rdg = NULL;
  Icon *ic = NULL;

  Window *win = Window::factory ("radio", ELM_WIN_BASIC);
  win->setTitle ("Radios");
  win->setAutoDel (true);

  Background *bg = Background::factory (*win);
  win->addObjectResize (*bg);
  bg->setWeightHintSize (EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
  bg->show ();

  Box *bx = Box::factory (*win);
  win->addObjectResize (*bx);
  bx->setWeightHintSize (EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
  bx->show ();

  ic = Icon::factory (*win);
  ic->setFile (searchPixmapFile ("elementaryxx/logo_small.png"));
  ic->setAspectHintSize (EVAS_ASPECT_CONTROL_VERTICAL, Size (1, 1));
  rd = Radio::factory (*win);
  rd->setWeightHintSize (EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
  rd->setAlignHintSize (EVAS_HINT_FILL, 0.5);
  rd->setText ("Icon sized to radio");
  rd->setContent (*ic);
  bx->packEnd (*rd);
  rd->show ();
  ic->show ();
  rdg = rd;

  ic = Icon::factory (*win);
  ic->setFile (searchPixmapFile ("elementaryxx/logo_small.png"));
  ic->setNoScale (true);
  rd = Radio::factory (*win);
  rd->setStateValue (1);
  rd->addGroup (*rdg);
  rd->setText ("Icon no scale");
  rd->setContent (*ic);
  bx->packEnd (*rd);
  rd->show ();
  ic->show ();

  rd = Radio::factory (*win);
  rd->setStateValue (2);
  rd->addGroup (*rdg);
  rd->setText ("Label Only");
  bx->packEnd (*rd);
  rd->show ();

  rd = Radio::factory (*win);
  rd->setStateValue (3);
  rd->addGroup (*rdg);
  rd->setText ("Disabled");
  rd->setDisabled (true);
  bx->packEnd (*rd);
  rd->show ();

  ic = Icon::factory (*win);
  ic->setFile (searchPixmapFile ("elementaryxx/logo_small.png"));
  ic->setNoScale (true);
  rd = Radio::factory (*win);
  rd->setStateValue (4);
  rd->addGroup (*rdg);
  rd->setContent (*ic);
  bx->packEnd (*rd);
  rd->show ();
  ic->show ();

  ic = Icon::factory (*win);
  ic->setFile (searchPixmapFile ("elementaryxx/logo_small.png"));
  ic->setNoScale (true);
  rd = Radio::factory (*win);
  rd->setStateValue (5);
  rd->addGroup (*rdg);
  rd->setContent (*ic);
  rd->setDisabled (true);
  bx->packEnd (*rd);
  rd->show ();
  ic->show ();

  rdg->setValue (2);

  win->show ();
}

File: BINDINGS/python/python-elementary/tests/test_radio.py
#!/usr/bin/env python
import os
import elementary
import edje
import ecore
import evas

#----- Radio -{{{-
def radio_clicked(obj, it):
    win = elementary.Window("radio", elementary.ELM_WIN_BASIC)
    win.title_set("Radio test")
    win.autodel_set(True)

    bg = elementary.Background(win)
    win.resize_object_add(bg)
    bg.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
    bg.show()

    bx = elementary.Box(win)
    bx.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
    win.resize_object_add(bx)
    bx.show()

    ic = elementary.Icon(win)
    ic.file_set('images/logo_small.png')
    ic.size_hint_aspect_set(evas.EVAS_ASPECT_CONTROL_VERTICAL, 1, 1)
    rd = elementary.Radio(win)
    rd.state_value_set(0)
    rd.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
    rd.size_hint_align_set(evas.EVAS_HINT_FILL, 0.5)
    rd.label_set("Icon sized to radio")
    rd.icon_set(ic)
    bx.pack_end(rd)
    rd.show()
    ic.show()
    rdg = rd

    ic = elementary.Icon(win)
    ic.file_set('images/logo_small.png')
    ic.scale_set(0, 0)
    rd = elementary.Radio(win)
    rd.state_value_set(1)
    rd.group_add(rdg)
    rd.label_set("Icon no scale")
    rd.icon_set(ic)
    bx.pack_end(rd)
    rd.show()
    ic.show()

    rd = elementary.Radio(win)
    rd.state_value_set(2)
    rd.group_add(rdg)
    rd.label_set("Label Only")
    bx.pack_end(rd)
    rd.show()

    rd = elementary.Radio(win)
    rd.state_value_set(3)
    rd.group_add(rdg)
    rd.label_set("Disabled")
    rd.disabled_set(True)
    bx.pack_end(rd)
    rd.show()

    ic = elementary.Icon(win)
    ic.file_set('images/logo_small.png')
    ic.scale_set(0, 0)
    rd = elementary.Radio(win)
    rd.state_value_set(4)
    rd.group_add(rdg)
    rd.icon_set(ic)
    bx.pack_end(rd)
    rd.show()
    ic.show()

    ic = elementary.Icon(win)
    ic.file_set('images/logo_small.png')
    ic.scale_set(0, 0)
    rd = elementary.Radio(win)
    rd.state_value_set(5)
    rd.group_add(rdg)
    rd.icon_set(ic)
    rd.disabled_set(True)
    bx.pack_end(rd)
    rd.show()
    ic.show()

    rdg.value_set(2)

    win.show()
# }}}

#----- Main -{{{-
if __name__ == "__main__":
    elementary.init()

    radio_clicked(None, None)

    elementary.run()
    elementary.shutdown()
# }}}
# vim:foldmethod=marker
